packages/react/src/Dialog/Popup/DialogPopup.tsx

```diff
@@ -114,6 +114,7 @@ const DialogPopup = React.forwardRef(function DialogPopup(
 
   return (
     <FloatingPortal root={container}>
+      {modal && mounted && <FloatingOverlay />}
```

Comment:
What about `z-index`?

Reply:
Yeah, right, this is going to be tricky. We could figure out the z-index of the popup and set this one lower, but this will be unreliable in some cases. We might have to expose the `className` of this part, or the whole part itself, to be consistent with the rest of the API.

Reply:
Is this another problem solved by a separate Portal part? 😅

Reply:
Not really, unless I'm missing something. Having an explicit Portal wouldn't help determining the right z-index of the internal backdrop.
Perhaps a better solution is not rendering the internal backdrop, but adding the `inert` attribute to all elements except the popup?

Reply:
> Perhaps a better solution is not rendering the internal backdrop, but adding the inert attribute to all elements except the popup?

This could work. `guards={false}` on `FloatingFocusManager` does this, but it allows escaping to the browser address bar of course, which we didn't want to allow. Guess this is relevant: https://github.com/floating-ui/floating-ui/discussions/3098

The other option is to get the computed `z-index` style of the `Popup` and apply the `z-index` to be just one less on `FloatingOverlay`?